### PR TITLE
Enable AVX2 and AVX-512 implementations of Keccak

### PR DESCRIPTION
**Pull request by nbgl**

### CODE REVIEW

File: keccak/src/avx2split.rs
Comment (nbgl) [inline keccak/src/avx2split.rs:1]: This is an alternative implementation for AVX2, but I'm deleting it since it's slower than the naive approach.

**Review by unzvfu — APPROVED** [review]
Looks good.

File: keccak/src/avx2.rs [hunk 1]
```diff
@@ -301,27 +301,24 @@ fn round(i: usize, state: [__m256i; 25]) -> [__m256i; 25] {
...
 
-fn keccak_perm(buf: &mut [__m256i; 25]) {
-    let mut state = *buf;
+fn keccak_perm(buf: &mut [[u64; VECTOR_LEN]; 25]) {
+    let mut state: [__m256i; 25] = unsafe { transmute_copy(buf) };
     for i in 0..24 {
         state = round(i, state);
     }
-    *buf = state;
+    *buf = unsafe { transmute_copy(&state) };
```
Comment (rjeli) [inline keccak/src/avx2.rs:309]: FWIW, [nomicon](https://doc.rust-lang.org/nomicon/transmutes.html) really doesn't like `transmute_copy`. but I'm not sure how much I trust it these days, and it seems fine to me...

Comment (nbgl) [reply keccak/src/avx2.rs:309]: fair! i really should just have done a `transmute(state)` here

**Pull request merged**